qiskit/pulse/configuration.py

```diff
@@ -14,11 +14,51 @@
 Configurations for pulse experiments.
 """
 from typing import Dict, Union, Tuple, Optional
+import numpy as np
 
 from .channels import PulseChannel, DriveChannel, MeasureChannel
 from .exceptions import PulseError
 
 
+def _assert_nested_dict_equal(a, b):
+    if len(a) != len(b):
+        return False
+    for key in a:
+        if key in b:
+            if isinstance(a[key], dict):
+                if not _assert_nested_dict_equal(a[key], b[key]):
+                    return False
+            elif isinstance(a[key], np.ndarray):
+                if not np.all(a[key] == b[key]):
+                    return False
+            else:
+                if a[key] != b[key]:
+                    return False
+        else:
+            return False
+    return True
+
+
+def _get_hash(obj: dict):
```

Comment:
This looks good to me, but following @nkanazawa1989's comment I think just removing hashing might be better since we can't think of a good use case and less code to maintain is always better.